core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4216,6 +4229,24 @@ public HttpResponse doSafeRestart(StaplerRequest req) throws IOException, Servle
         return HttpResponses.redirectToDot();
     }
 
+    private boolean isRestClientAndPost(@CheckForNull StaplerRequest req) {
```

Comment:
possibly may be better to look to see if it was using token based authentication?

Reply:
[this](https://github.com/jenkinsci/jenkins/blob/8bea0c952621c0031168bd1ea3123917e450bac4/core/src/main/java/jenkins/security/BasicHeaderApiTokenAuthenticator.java#L48) may help :)

Reply:
Unfortunately, using the token as a sign of a REST client will not work for every case, like when you have no security enable.

The different endpoints strategy seems to be the only one that will satisfy everybody.